joladnijo/settings.py

```diff
@@ -141,24 +148,38 @@
 # Static files (CSS, JavaScript, Images)
 # https://docs.djangoproject.com/en/3.2/howto/static-files/
 
-STATIC_URL = '/static/'
+STATIC_URL = "/static/"
 
 # Default primary key field type
 # https://docs.djangoproject.com/en/3.2/ref/settings/#default-auto-field
 
-DEFAULT_AUTO_FIELD = 'django.db.models.BigAutoField'
+DEFAULT_AUTO_FIELD = "django.db.models.BigAutoField"
 
+# REST Framework
 REST_FRAMEWORK = {
-    'DEFAULT_RENDERER_CLASSES': [
-        'djangorestframework_camel_case.render.CamelCaseJSONRenderer',
-        'djangorestframework_camel_case.render.CamelCaseBrowsableAPIRenderer',  # TODO: törölni majd
-    ],
-    'DEFAULT_PARSER_CLASSES': [
-        'djangorestframework_camel_case.parser.CamelCaseJSONParser',
-    ],
-    'DEFAULT_PERMISSION_CLASSES': [
-        'rest_framework.permissions.IsAuthenticatedOrReadOnly',
-    ],
+    "DEFAULT_PERMISSION_CLASSES": (
+        "rest_framework.permissions.IsAuthenticated",
+        "rest_framework.permissions.IsAuthenticatedOrReadOnly",
+    ),
+    "DEFAULT_AUTHENTICATION_CLASSES": (
+        "rest_framework_jwt.authentication.JSONWebTokenAuthentication",
+        "rest_framework.authentication.SessionAuthentication",
+        "rest_framework.authentication.BasicAuthentication",
+    ),
+    "DEFAULT_RENDERER_CLASSES": (
+        "djangorestframework_camel_case.render.CamelCaseJSONRenderer",
+        "djangorestframework_camel_case.render.CamelCaseBrowsableAPIRenderer",  # TODO: törölni majd
+    ),
+    "DEFAULT_PARSER_CLASSES": ("djangorestframework_camel_case.parser.CamelCaseJSONParser",),
+}
+
+JWT_AUTH = {
+    "JWT_PAYLOAD_GET_USERNAME_HANDLER": "joladnijo.auth0.jwt_get_username_from_payload_handler",
+    "JWT_DECODE_HANDLER": "joladnijo.auth0.jwt_decode_token",
+    "JWT_ALGORITHM": os.environ.get('JWT_ALGORITHM', 'RS256'),
+    "JWT_AUDIENCE": os.environ.get('JWT_AUDIENCE'),
+    "JWT_ISSUER": os.environ.get('JWT_ISSUER'),
+    "JWT_AUTH_HEADER_PREFIX": "Bearer",
```

Comment:
Ezt nem akartad használni a `joladnijo/auth0.py`-ban?
Azoknál a kulcsoknál, amiknek nincs defaul értéke, már itt dobhatnánk egy kivételt, ha nincsenek beállítva (akár sima `KeyError`-t).

Reply:
Ez itt a Django REST Framework beépített authorizációs cuccaihoz kell (lehet, hogy nem fogjuk végül használni őket, ha mindenhol custom permission-kezelés lesz, de egyelőre maradjon). Beteszem majd ide az ellenőrzéseket.